.ci/test_r_package.sh

```diff
@@ -186,7 +195,7 @@ if grep -q -R "WARNING" "$LOG_FILE_NAME"; then
     exit -1
 fi
 
-ALLOWED_CHECK_NOTES=2
+ALLOWED_CHECK_NOTES=0
```

Comment:
Great to see that we can simplify NOTEs checking by just some `export`s!

I think we can now change this to simple `if grep -q -R "NOTE ..."`, can't we?

Reply:
oh sure, I can do that

Reply:
while we're thinking about it, I think we should add a fix for https://github.com/microsoft/LightGBM/issues/3616 in this PR too:

```shell
if grep -q -R "ERROR" "$LOG_FILE_NAME"; then
    echo "ERRORs have been found by R CMD check!"
    exit -1
fi
```